edtLHD/src/main/java/fr/univtln/lhd/model/entities/dao/slots/SlotDAO.java

```diff
@@ -21,64 +22,102 @@
-
-    private static final String GETALL_STMT = "SELECT * FROM SLOTS";
-
+    private static final String GET_SLOT_FROM_PROFESSOR_STMT = "SELECT " + FORMATTED_SLOT + " FROM  slots JOIN professor_slot on id_slot = slots.id WHERE id_professor= (?)";//NOSONAR
+    private static final String GET_SLOT_FROM_GROUP_STMT = "SELECT " + FORMATTED_SLOT + " FROM  slots JOIN group_slot on id_slot = slots.id WHERE id_group= (?)";//NOSONAR
+    private static final String GETALL_STMT = "SELECT * FROM SLOTS";//NOSONAR
     private static final String DELETE_STMT = "DELETE FROM slots WHERE id=(?)";
 
-    public static SlotDAO getInstance () {
+    private SlotDAO() {
+    }
+
+    public static SlotDAO getInstance() {
         return new SlotDAO();
     }
 
+    /**
+     * A wrapper for selection to avoid result set duplication
+     * @param rs a ResultSet stemming from a query
+     * @return the Slot to select
+     * @throws SQLException caller to manage
+     * @throws IdException caller to manage
+     */
+    private Slot getFromResultSet(final ResultSet rs) throws SQLException, IdException {
+        final long id = (rs.getLong("ID"));
+        final Slot result = Slot.getInstance(
+                Slot.SlotType.valueOf(rs.getString("TYPE")),
+                ClassroomDAO.getInstance().get(rs.getLong("CLASSROOM")).orElseThrow(SQLException::new),
+                SubjectDAO.getInstance().get(rs.getLong("SUBJECT")).orElseThrow(SQLException::new),
+                new HashSet<>(GroupDAO.getInstance().getGroupOfSlot(id)), //the returned list is unmodifiable, requiring a copy.
+                ProfessorDAO.of().getProfessorOfSlots(id),
+
+                Interval.of(Instant.parse(rs.getString("begin")),
+                        Instant.parse(rs.getString("end"))),
+                rs.getString("MEMO")
+        );
+        result.setId(id);
+        return result;
+    }
 
     /**
-     * Getter for one Slot
-     *
-     * @param id numerical long identifier for getting the Slot
-     * @return May return one Slot instance
+     * Get a SQL query from parameters with a date-based filter
+     * @param id the id pf the slot to get
+     * @param date the date to base the filter upon
+     * @param query SQL query to be executed with passed parameters
+     * @return A set of corresponding slot entities
+     * @throws SQLException if the query goes wrong
      */
-    @Override
-    public Optional<Slot> get ( final long id ) throws SQLException {
-        Optional<Slot> result = Optional.empty();
+    private Set<Slot> getList(final long id, final Instant date, @NonNull final String query) throws SQLException {
+        Slot result;
+        Set<Slot> slotSet = new HashSet<>();
         try (Connection conn = Datasource.getConnection();
-             PreparedStatement stmt = conn.prepareStatement(GET_STMT)
+             PreparedStatement stmt = conn.prepareStatement(query)
```

Comment:
The query is expected to be of the form 

"... WHERE<Condition on id> and <Condition on Date> ..."